File: src/javaAlgorithm1/Solution2.java
package javaAlgorithm1;

public class Solution2 {
	public static void main(String[] args) {
		
		System.out.println(solution(626331));
		
	}


	public static int solution(int num) {
		int answer = 0;
		int n= num;
		int a= 0;
		while(n!=1 && n!=-1) {
			if(n%2==0) {
				n = n/2;

			}
			else if(n%2==1) {
				n=n*3+1;
			}
			
			else if(answer>500) {
				answer = -1;
				break;
			}
			answer += 1;
			
		}

		
		
		return answer;
	}
}



//
//1-1. 입력된 수가 짝수라면 2로 나눕니다. 
//1-2. 입력된 수가 홀수라면 3을 곱하고 1을 더합니다.
//2. 결과로 나온 수에 같은 작업을 1이 될 때까지 반복합니다.

File: src/techCamp4th/Solution2.js
function solution(leave,day,holidays) {
    if (day =='MON'){
        holidays = holidays.concat([6,7,13,14,20,21,27,28]);
    } else if(day == 'TUE'){
        holidays = holidays.concat([5,6,12,13,19,20,26,27]);
    } else if (day == 'WED'){
        holidays = holidays.concat([4,5,11,12,18,19,25,26]);
    } else if (day == 'THU'){
        holidays = holidays.concat([3,4,10,11,17,18,24,25]);
    } else if (day =='FRI'){
        holidays = holidays.concat([2,3,9,10,16,17,23,24,30]);
    } else if (day =='SAT'){
        holidays = holidays.concat([1,2,8,9,15,16,22,23,29,30]);
    } else {
        holidays = holidays.concat([1,7,8,14,15,21,22,28,29]);
    }
    holidays.sort(function(a, b)  {
        if(a > b) return 1;
        if(a === b) return 0;
        if(a < b) return -1;
    });
    var answer = 0;
    var holidayIndex = 0;
    var maxDay = 0;

    for (var i = 1 ; i <= 30 ; i++){
        var remain = leave;

        while (holidays[holidayIndex]<i){
            holidayIndex++;
        }
        j=i;
        while (remain>=0){

            if (j == holidays[holidayIndex]){
                maxDay++;
                if (holidays[holidayIndex]==holidays[holidayIndex+1]){
                    holidayIndex++;
                }
                holidayIndex++;
            } else {
                remain--;
                maxDay++;
            }
            j++;
        }
        if (answer<maxDay){
            answer=maxDay;
        }
        holidayIndex=0;
        maxDay=0;
    }
    return answer;
}

console.log(solution(4,'MON',[6, 21, 23, 27, 28]));